### PR TITLE
apply new *-icon-for-dir API and use *-dired-dir-face

### PR DESCRIPTION
**Pull request by conao3**
This PR solves dir face issue which is mentioned at https://github.com/jtbm37/all-the-icons-dired/pull/31#issuecomment-593062849 .
But this PR depends PR sent all-the-icons (domtronn/all-the-icons.el/pull/181).
So, this PR send as WIP PR until the PR merged.

### CODE REVIEW

**Comment by conao3** [comment]
Screenshot_2020-03-01_23-53-10
- left is dired before #24 merged
- middle is dired after #24 merged
- right is dired after this PR merged (and domtronn/all-the-icons.el/pull/181 merged)

File: all-the-icons-dired.el [hunk 1]
```diff
@@ -82,7 +82,7 @@
       (let ((file (dired-get-filename 'verbatim t)))
         (when file
           (let ((icon (if (file-directory-p file)
-                          (all-the-icons-icon-for-dir file nil "")
+                          (all-the-icons-icon-for-dir file :face 'all-the-icons-dired-dir-face)
```
Comment (seagle0128) [inline all-the-icons-dired.el]: I suggest to add `:v-adjust -0.05` for the alignment.

**Comment by conao3** [comment]
pass `all-the-icons-dired-v-adjust` to `all-the-icons-icon-for-dir` as override arg.

[Note]
Do we need to adjust `:v-adjast` or `:font` in each package like this?
If we need to make these tweaks, we can modify upstream and get its contributions in many packages.

**Comment by seagle0128** [comment]
> Do we need to adjust `:v-adjast` or `:font` in each package like this?
> If we need to make these tweaks, we can modify upstream and get its contributions in many packages.

@conao3  Actually I submitted the PR https://github.com/domtronn/all-the-icons.el/pull/173. The maintainer thinks it's hard to adapt all fonts. So, the `args` is necessary here IMO.

**Comment by conao3** [comment]
OK, thanks!

**Comment by conao3** [comment]
`all-the-icons` upstream PR is merged :tada:
This PR could be mergeable!

**Comment by jtbm37** [comment]
@conao3 Thanks!

**Pull request merged**